components/layout/display_list_builder.rs

```diff
@@ -1644,18 +1638,24 @@ impl BlockFlowDisplayListBuilding for BlockFlow {
                 StackingContextCreationMode::PseudoFloat
             };
 
-            let mut stacking_context =
-                    self.fragment.create_stacking_context(stacking_context_id,
-                                                          &self.base,
-                                                          ScrollPolicy::Scrollable,
-                                                          creation_mode);
-            let (mut floating, mut positioned) = child_contexts.into_iter().partition(|context| {
-                context.context_type == StackingContextType::PseudoFloat
-            });
+            let stacking_context_index = contexts.len();
+            contexts.push(self.fragment.create_stacking_context(stacking_context_id,
+                                                                &self.base,
+                                                                ScrollPolicy::Scrollable,
+                                                                creation_mode));
+
+            let mut floating = vec![];
+            for child_context in child_contexts.into_iter() {
+                if child_context.context_type == StackingContextType::PseudoFloat {
+                    // Floating.
+                    floating.push(child_context)
+                } else {
+                    // Positioned.
+                    contexts.push(child_context)
+                }
+            }
 
-            stacking_context.children.append(&mut floating);
-            contexts.push(stacking_context);
-            contexts.append(&mut positioned);
+            contexts[stacking_context_index].children = floating;
```

Comment:
I wonder if there is some other way to do this without copying. I'm a little wary of this because it means that when you dump the display list, the output will not accurately reflect the structure of the stacking contexts. Is a contexts.append(...) here so expensive now that stacking contexts are boxed?